src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -563,7 +565,8 @@ private static ImageProvider createImageProvider(JlinkConfiguration config,
 
         Map<String, Path> mods = cf.modules().stream()
             .collect(Collectors.toMap(ResolvedModule::name, JlinkTask::toPathLocation));
-        return new ImageHelper(cf, mods, config.getByteOrder(), retainModulesPath, ignoreSigning);
+        return new ImageHelper(cf, mods, config.getByteOrder(), retainModulesPath, ignoreSigning,
```

Comment:
Also we can move the logic of determining the endianess of the target platform out of `ImageHelper`.   Evaluate the target platform and endianness here and pass to `ImageHelper`.

```suggestion
        // endian is the value specified via --endian and a new parameter to createImageProvider 
        Platform targetPlatform = targetPlatform(cf, modsPaths);
        ByteOrder targetOrder = endian != null ? endian : targetPlatform.arch().byteOrder();
        return new ImageHelper(cf, mods, targetOrder, retainModulesPath, ignoreSigning,
```

Reply:
I've updated the PR to use this suggestion. With this suggested change, the code is much more cleaner and I now just pass the `targetPlatform` to the `ImageHelper` constructor.